src/services/ApiService.ts

```diff
@@ -185,58 +185,69 @@ export class ApiService {
+                        // Remove the invalid preference from the preferences object
+                        delete preferences[errorKey];
+                        this.clearPreferences([errorKey]);
+                        deletedKeys.push(errorKey);
                     }
                 }
+
+                if (deletedKeys.length > 0) {
+                    // Show an alert to the user about the deleted preferences
+                    AppStore.Instance.alertStore.showAlert(`Deleted invalid preferences: ${deletedKeys.join(", ")}`);
+                }
             }
         }
         return preferences;
     };
 
     private upgradePreferences = (preferences: any) => {
         // Upgrade to V2 if required
-        if (preferences?.version && preferences.version === 1) {
-            if (preferences.astColor || preferences.astColor === 0) {
-                switch (preferences.astColor) {
-                    case 0:
-                        preferences.astColor = "auto-black";
-                        break;
-                    case 1:
-                        preferences.astColor = "auto-white";
-                        break;
-                    case 2:
-                        preferences.astColor = "auto-red";
-                        break;
-                    case 3:
-                        preferences.astColor = "auto-forest";
-                        break;
-                    case 4:
-                        preferences.astColor = "auto-blue";
-                        break;
-                    case 5:
-                        preferences.astColor = "auto-turquoise";
-                        break;
-                    case 6:
-                        preferences.astColor = "auto-violet";
-                        break;
-                    case 7:
-                        preferences.astColor = "auto-gold";
-                        break;
-                    case 8:
-                        preferences.astColor = "auto-gray";
-                        break;
-                    default:
-                        preferences.astColor = "auto-blue";
-                        break;
-                }
-            }
-            preferences.version = 2;
-            this.setPreferences(preferences);
+        if (preferences["version"] === 1) {
+            // Convert preferences[PreferenceKeys.WCS_OVERLAY_AST_COLOR] from a number in version 1 to a string in version 2
+            // default to "auto-blue" if the value is not in the AST_COLORS map
+            enum ASTColors {
+                black,
+                white,
+                red,
+                forest,
+                blue,
+                turquoise,
+                violet,
+                gold,
+                gray
+            }
+            const astColorKey = PreferenceKeys.WCS_OVERLAY_AST_COLOR;
+            const color = preferences[astColorKey] in ASTColors ? ASTColors[preferences[astColorKey]] : "blue";
```

Comment:
I've realised that `in` is not sufficient here -- you also have to check that `preferences[astColorKey]` is a number. Otherwise a string value which is a valid enum key will result in the reverse lookup (and we'll end up with `auto-3`, etc.).

We should also not rely on implicit numbering in the enum, because the numeric values have meaning -- it would be safer to add them explicitly.